api/src/main/java/com/epam/pipeline/manager/cluster/writer/XlsMonitoringStatsWriter.java

```diff
@@ -0,0 +1,187 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.manager.cluster.writer;
+
+import com.epam.pipeline.common.MessageConstants;
+import com.epam.pipeline.common.MessageHelper;
+import com.epam.pipeline.entity.cluster.monitoring.MonitoringStats;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.poi.hssf.usermodel.HSSFFormulaEvaluator;
+import org.apache.poi.hssf.util.CellReference;
+import org.apache.poi.openxml4j.exceptions.InvalidFormatException;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellStyle;
+import org.apache.poi.ss.usermodel.CellType;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.ss.usermodel.WorkbookFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@SuppressWarnings("checkstyle:MagicNumber")
+public class XlsMonitoringStatsWriter extends AbstractMonitoringStatsWriter {
+
+    private static final String SCALED_DATA_SHEET = "SCALED_DATA";
+    private static final String DISK_DATA_SHEET = "DISKS_SUMMARY";
+    private static final String RAW_DATA_SHEET = "DATA";
+    private static final String CPU_CONVERSION_FORMULA = "DATA!%c%d/100";
+    private static final String MEM_CONVERSION_FORMULA = "DATA!E%d*DATA!%c%d/100/1073741824";
+    private static final String NET_CONVERSION_FORMULA = "DATA!%s/1048576";
+    private static final String DISK_USED_FORMULA = "DATA!%s/1073741824*DATA!%s/100";
+    private static final String DISK_FREE_FORMULA = "DATA!%s/1073741824*(100-DATA!%s)/100";
+    private static final Character CPU_AVG_COLUMN = 'C';
+    private static final Character CPU_MAX_COLUMN = 'D';
+    private static final Character MEM_AVG_COLUMN = 'F';
+    private static final Character MEM_MAX_COLUMN = 'G';
+    private static final long BYTES_IN_GB = 1L << 30;
+    private static final String DISK_NAME_TEMPLATE = "%s[%.2fGb]";
+    private static final String NUMERIC_CELL_PRECISION_FORMAT = "0.00";
+
+    private final String templatePath;
+    private final MessageHelper messageHelper;
+
+    public XlsMonitoringStatsWriter(final @Value("${monitoring.stats.export.xls.template}") String templatePath,
+                                    final MessageHelper messageHelper) {
+        this.templatePath = templatePath;
+        this.messageHelper = messageHelper;
+    }
+
+    @Override
+    public synchronized InputStream convertStatsToFile(final List<MonitoringStats> stats) {
+        try (Workbook wb = WorkbookFactory.create(new File(templatePath), null, true);
```

Comment:
I assume a proper error message would be nice in case `templatePath` is either blank or null.

Reply:
Done